_posts/2016-07-14-official-docker-image.md

```diff
@@ -0,0 +1,170 @@
+this project.
+
+We always had a Docker image available, but that was oriented toward developers,
+hence using it in production would have been sub-optimal.
+
+The next paragraphs illustrate how the image has been created and the topics we
+focused on.
+
+## Security
+
+SUSE's containers team cares about security, hence we made the following
+decisions:
+
+  * This image enforces the usage of HTTPS to access Portus.
+  * Portus is installed from an RPM package, this ensures the final image will
+    have only the runtime dependency; no build time dependency is ever installed.
+  * The default user for this image is the `wwwrun` one, which is created by the
+    Apache package.
+  * The root account is locked, that means it's not possible to switch user via
+    `su` or `sudo`.
+
+SUSE's containers team is constantly working to automate the release process
+of Portus and of this image to ensure it stays up-to-date and secure.
+
+When deploying this image make sure to add all the required keys and
+certificates at runtime instead of adding them to a Docker image.
+
+## Anatomy of the image
+
+The image is based on the latest stable release of openSUSE (right now Leap 42.1)
+and installs Portus using the RPM package built by SUSE's containers team inside
+of the [Open Build Service](https://build.opensuse.org/project/subprojects/Virtualization:containers:Portus)
+top level project and subprojects (one subproject per Portus branch).
+
+### Exposed ports
+
+This image uses [Phusion Passenger](https://www.phusionpassenger.com/) in
+conjunction with Apache to run Portus.
+
+The container will expose the following ports:
+
+  * `443`: this is used to serve the website over HTTPS.
+  * `80`: most of the requests sent to this unencrypted channel will be
+    automatically redirected to the HTTPS channel.
+    The redirection is made using Apache's [mod_rewrite](https://httpd.apache.org/docs/current/mod/mod_rewrite.html).
+
+The redirection from port `80` to port `443` applies to all the common requests sent
+to Portus. The only requests that are **not** redirected are the ones sent to
+the `v2/webhooks/events` endpoint. This is the endpoint used by the Docker
+Registry to send notifications to Portus.
+
+Sending Registry's notification to the HTTPS endpoint would require Portus'
+certificate to be added to the Docker container running the registry. This is
+not hard to be done, but involves some changes to the official Registry image
+maintained by Docker Inc (or some hacks like a custom entrypoint added via a volume).
+
+### The init script
+
+As mentioned before, this Docker image has a init script which takes care of the
+following actions:
+
+  1. Setup the database required by Portus
+  2. Import all the `.crt` located under `/certificates`
+  3. Start Apache
+
+The next sections will provide more details about these steps.
+
+### Volumes
+
+Portus' state is stored inside of a MariaDB database. This makes this Docker
```

Comment:
I would not use "this" two-times in a row. Rather "This make the Docker image stateless."